frontend/src/container/GuestUpload.jsx

```diff
@@ -114,17 +116,19 @@ class GuestUpload extends React.Component {
 
     if (!Array.isArray(newFileList) || (newFileList.length === 0)) return
 
-    const alreadyUploadedList = newFileList.filter(newFile => state.fileToUploadList.some(stateFile => stateFile.name === newFile.name))
-    if (alreadyUploadedList.length) {
+    const droppedFileUploadList = newFileList.map(file => createFileUpload(file))
+    const alreadyPresentList = droppedFileUploadList.filter(newFileUpload => isFileUploadInList(newFileUpload, state.fileUploadList))
+    const newFileUploadList = droppedFileUploadList.filter(newFileUpload => !isFileUploadInList(newFileUpload, state.fileUploadList))
+    if (alreadyPresentList.length) {
       GLOBAL_dispatchEvent({
         type: CUSTOM_EVENT.ADD_FLASH_MSG,
         data: {
           msg: (
             <div>
-              {props.t('Files already uploaded:')}
+              {props.t('Files already added:')}
```

Comment:
Well, technically, this would be "A file with the same name is already there", since the user could try to upload two different files with the same file name (it already happened to me).
But this is out of scope of this PR probably